infra/feast-operator/internal/controller/services/util.go

```diff
@@ -0,0 +1,94 @@
+package services
+
+import (
+	"github.com/feast-dev/feast/infra/feast-operator/api/feastversion"
+	feastdevv1alpha1 "github.com/feast-dev/feast/infra/feast-operator/api/v1alpha1"
+)
+
+func IsLocalRegistry(featureStore *feastdevv1alpha1.FeatureStore) bool {
+	appliedServices := featureStore.Status.Applied.Services
+	return appliedServices != nil && appliedServices.Registry != nil && appliedServices.Registry.Local != nil
+}
+
+func ApplyDefaultsToStatus(cr *feastdevv1alpha1.FeatureStore) {
+	cr.Status.FeastVersion = feastversion.FeastVersion
+	applied := cr.Spec.DeepCopy()
+	if applied.Services == nil {
+		applied.Services = &feastdevv1alpha1.FeatureStoreServices{}
+	}
+
+	// default to registry service deployment
+	if applied.Services.Registry == nil {
+		applied.Services.Registry = &feastdevv1alpha1.Registry{
+			Local: &feastdevv1alpha1.LocalRegistryConfig{
+				Persistence: &feastdevv1alpha1.RegistryPersistence{
+					FilePersistence: &feastdevv1alpha1.RegistryFilePersistence{
+						Path: DefaultRegistryPath,
+					},
+				},
+			},
+		}
+	}
+	// if remote registry not set, proceed w/ local registry defaults
+	if applied.Services.Registry.Remote == nil {
+		// if local registry not set, apply an empty pointer struct
+		if applied.Services.Registry.Local == nil {
+			applied.Services.Registry.Local = &feastdevv1alpha1.LocalRegistryConfig{
+				Persistence: &feastdevv1alpha1.RegistryPersistence{
+					FilePersistence: &feastdevv1alpha1.RegistryFilePersistence{
+						Path: DefaultRegistryPath,
+					},
+				},
+			}
+		}
```

Comment:
no need to file persistence to be applied twice here. instead, need to cover all bases by checking each pointer object separately, but persistence is ultimately only set once -
```suggestion
	// default to registry service deployment
	if applied.Services.Registry == nil {
		applied.Services.Registry = &feastdevv1alpha1.Registry{},
	}
	// if remote registry not set, proceed w/ local registry defaults
	if applied.Services.Registry.Remote == nil {
		// if local registry not set, apply an empty pointer struct
		if applied.Services.Registry.Local == nil {
			applied.Services.Registry.Local = &feastdevv1alpha1.LocalRegistryConfig{},
		}
		if applied.Services.Registry.Local.Persistence == nil {
			applied.Services.Registry.Local.Persistence = &feastdevv1alpha1.RegistryPersistence{
					FilePersistence: &feastdevv1alpha1.RegistryFilePersistence{
						Path: DefaultRegistryPath,
					},
				},
			}
		}
```

Reply:
this is pertinent in case the user defines `spec.services.registry.local: {}` for example. we'd still need to set some persistence defaults.

Reply:
we prob need to incorporate an `if applied.Services.Registry.Local.Persistence.FilePersistence != nil && len(applied.Services.Registry.Local.Persistence.FilePersistence) == 0` check in here too right? in that scenario we'd `applied.Services.Registry.Local.Persistence.FilePersistence.Path = DefaultRegistryPath`

Reply:
@dmartinol i think we still need to address an empty `Path` or `Type` depending on the feast service type. see the comment directly above this one.

Reply:
done